HeterogeneousCore/SonicTriton/interface/TritonUtils.h

```diff
@@ -0,0 +1,26 @@
+#ifndef HeterogeneousCore_SonicTriton_TritonUtils
+#define HeterogeneousCore_SonicTriton_TritonUtils
+
+#include <string>
+#include <vector>
+
+#include "request_grpc.h"
+
+namespace TritonUtils {
+
+  using Error = nvidia::inferenceserver::client::Error;
+
+  template <typename T>
+  std::string printVec(const std::vector<T>& vec, const std::string& delim = ", ");
+
+  //helper to turn triton error into exception
+  void wrap(const Error& err, const std::string& msg);
+
+  //helper to turn triton error into warning
+  bool warn(const Error& err, const std::string& msg);
```

Comment:
Just a thought, `std::string_view` argument would allow avoiding one intermediate `std::string` when the call has a string literal (same for `wrap()`).